kernel/dev/tty/tty.cpp

```diff
@@ -142,11 +153,36 @@ void putchar(char c) {
                 }
                 ansi_state = Normal;
                 ansi_val = 0;
+            } else if (c == 'H' || c == 'f') { // Set cursor position attribute
+                PUSH_VAL(ansi_val);
+                // take action here
+                // iterate through all values
+                // the proper order is 'line (y);column (x)'
+                while (ansi_values_index > 0) {
+                    ansi_val = POP_VAL();
+                    // If we only have one item in the stack now (i.e. line is done)
+                    if (ansi_values_index == 1) {
```

Comment:
What happens if there were more or less than 2 values provided? Should it always accept them and just only look at the first two pushed?

Reply:
> What happens if there were more or less than 2 values provided? Should it always accept them and just only look at the first two pushed?

I think that would be easiest. It would be non-standard to provide more than 2 values, so the safest thing to do is just ignore them. The code will need a little tweaking to check for that, but it won't be too hard.

Reply:
It looks like it will already behave that way